applications/app/jobs/SitemapLifecycle.scala

```diff
@@ -0,0 +1,42 @@
+package jobs
+
+import common._
+import play.api.GlobalSettings
+import services.{VideoSiteMap, NewsSiteMap}
+
+trait SiteMapLifecycle extends GlobalSettings with ExecutionContexts {
+
+  override def onStart(app: play.api.Application) {
+    super.onStart(app)
+
+    Jobs.deschedule("SiteMap")
+    Jobs.schedule("SiteMap", "0 0/2 * * * ?") {
+      SiteMapJob.update()
+    }
+
+    AkkaAsync {
+      SiteMapJob.update()
+    }
+  }
+}
+
+object SiteMapJob extends ExecutionContexts with Logging {
+  case class SiteMapContent(
+    news: xml.NodeSeq,
+    video: xml.NodeSeq)
+
+  private val siteMapContent = AkkaAgent[SiteMapContent](SiteMapContent(Nil, Nil))
```

Comment:
is it better to give an empty sitemap before the job completes successfully or an error?